libdevcrypto/ECDHE.cpp

```diff
@@ -1,37 +0,0 @@
-/*
- This file is part of cpp-ethereum.
- 
- cpp-ethereum is free software: you can redistribute it and/or modify
- it under the terms of the GNU General Public License as published by
- the Free Software Foundation, either version 3 of the License, or
- (at your option) any later version.
- 
- cpp-ethereum is distributed in the hope that it will be useful,
- but WITHOUT ANY WARRANTY; without even the implied warranty of
- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- GNU General Public License for more details.
- 
- You should have received a copy of the GNU General Public License
- along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
- */
-/** @file ECDHE.cpp
- * @author Alex Leverington <nessence@gmail.com>
- * @date 2014
- */
-
-#include "ECDHE.h"
-
-using namespace dev;
-using namespace dev::crypto;
-
-void ECDHE::agree(Public const& _remote, Secret& o_sharedSecret) const
-{
-	if (m_remoteEphemeral)
```

Comment:
I have the feeling that this might be a protection against an attack where you retrieve the private key by performing multiple key agreements. If possible, I would like to keep it as an internal assertion, even (or especially) if it is never triggered by the current code.